index.js

```diff
@@ -10,19 +15,35 @@ const NativeCallDetectorAndroid = NativeModules.CallDetectionManagerAndroid
 var CallStateUpdateActionModule = require('./CallStateUpdateActionModule')
 BatchedBridge.registerCallableModule('CallStateUpdateActionModule', CallStateUpdateActionModule)
 
+const requestPermissionsAndroid = (permissionMessage) => {
+  PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE)
+    .then((gotPermission) => gotPermission
+      ? true
+      : PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE, permissionMessage)
+          .then((result) => result === PermissionsAndroid.RESULTS.GRANTED)
+    )
+
+export const PermissionDenied = 'PERMISSION DENIED'
 class CallDetectorManager {
 
     subscription;
     callback
-    constructor(callback) {
+    constructor(callback, permissionDeniedCallback = console.error, permissionMessage = {
+      title: 'Phone State Permission',
+      message: 'This app needs access to your phone state in order to react and/or to adapt to incoming calls.'
+    }) {
         this.callback = callback
         if (Platform.OS === 'ios') {
             NativeCallDetector && NativeCallDetector.startListener()
             this.subscription = new NativeEventEmitter(NativeCallDetector)
             this.subscription.addListener('PhoneCallStateUpdate', callback);
         }
         else {
-            NativeCallDetectorAndroid && NativeCallDetectorAndroid.startListener()
+            if(NativeCallDetectorAndroid) {
+              requestPermissionsAndroid(permissionMessage)
```

Comment:
Don't make permission mandatory. As few users might not be interested in reading the incoming phone number, they might be interested in just detecting the incoming call. Please make this new feature optional and not a mandatory task. Just take the argument from the constructor which will indicate if they want to read incoming number, based on that value you can decide if you want to show permission or not.

Reply:
Done.